flink-runtime/src/test/java/org/apache/flink/runtime/memory/MemoryManagerConcurrentModReleaseTest.java

```diff
@@ -97,6 +101,12 @@ public void run() {
                 try {
                     MemorySegment seg = toModify.remove(0);
                     toModify.add(seg);
+                    // if the test running time reaches TEST_TIMEOUT_MS, we should lower the
+                    // priority of the Modifier thread to let segment release completes sooner.
```

Comment:
nit: this code does not lower the thread priority, yield is just a hint to the scheduler that the current thread is willing to yield its current use of a processor.

Reply:
@davidradl , thanks for the comment. I agree `yield` depends on the OS implementation. I changed to `sleep(1)` to force yield the CPU resource.

Reply:
Thanks for the comment @davidradl !
I've updated the comments and changed to use `exponential sleep` to make it much reliable. Thanks.